src/multiplex/client.rs

```diff
@@ -173,18 +383,23 @@ where
     T::Ok: 'static + Send,
     T::Tag: Send,
 {
-    /// Construct a new [`Client`] over the given `transport`.
-    ///
-    /// If the Client errors, the error is dropped when `new` is used -- use `with_error_handler`
-    /// to handle such an error explicitly.
-    pub fn new(transport: T) -> Self where {
```

Comment:
Any particular reason to not continue to expose a `new` for users who just want the default `P` and error handler?

Reply:
I figured it was better to just expose one way to construct the client?  But I guess we could support all the older methods what do you think?

Reply:
Ah, no, I was thinking _just_ `new` for the super simple (but also very common) case. Anything else should go through the builder.